xwiki-platform-core/xwiki-platform-url/xwiki-platform-url-default/src/main/java/org/xwiki/url/internal/DefaultURLSecurityManager.java

```diff
@@ -223,10 +225,16 @@ public URI parseToSafeURI(String serializedURI) throws URISyntaxException, Secur
             if (matcher.matches()) {
                 String scheme = matcher.group(2);
                 String authority = matcher.group(4);
```

Comment:
I'm wondering if we should do something with the authority, too, according to https://url.spec.whatwg.org/#hosts-(domains-and-ip-addresses) there could be percent-encoding in the host, too.

Reply:
I hesitated to add it. I'll change that and cover it with test.